### ISSUE TITLE
Output figures to a nonexistent directory

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#1983]
Users may want to generate figures in another directory. If that directory doesn't exist, GMT reports errors. 

Classic mode example:
```
gmt psbasemap -R0/10/0/10 -JX10c -Baf > test.ps
gmt psconvert -A -P -Tf test.ps -Dplot
```
Error message:
```
sh: plot/psconvert_50948c.bb: No such file or directory
psconvert [ERROR]: System call [gs -q -dSAFER -dNOPAUSE -dBATCH -sDEVICE=bbox -DPSL_no_pagefill -dMaxBitmap=2147483647 -dUseFastColor=true 'test.ps' 2> 'plot/psconvert_50948c.bb'] returned error 256.
psconvert [ERROR]: Failed to remove plot/psconvert_50948c.bb! [remove error: No such file or directory]
```

Modern mode example:
```
gmt begin plot/map pdf
	gmt basemap -R0/10/0/10 -JX10c -Baf
gmt end show
```
```
gmt begin map pdf Dplot
	gmt basemap -R0/10/0/10 -JX10c -Baf
gmt end show
```

Error message:
```
GPL Ghostscript 9.50: **** Could not open the file plot/map.pdf .
**** Unable to open the initial device, quitting.
psconvert [ERROR]: System call [gs -q -dNOPAUSE -dBATCH -dNOSAFER -dPDFSETTINGS=/prepress -dDownsampleColorImages=false -dDownsampleGrayImages=false -dDownsampleMonoImages=false -dUseFlateCompression=true -dEmbedAllFonts=true -dSubsetFonts=true -dMonoImageFilter=/FlateEncode -dAutoFilterGrayImages=false -dGrayImageFilter=/FlateEncode -dAutoFilterColorImages=false -dColorImageFilter=/FlateEncode -dSCANCONVERTERTYPE=2 -dMaxBitmap=2147483647 -dUseFastColor=true -sDEVICE=pdfwrite  -g3285x3210 -r720 -sOutputFile='plot/map.pdf' '/Users/seisman/.gmt/sessions/gmt6.51095/psconvert_51098d.eps'] returned error 256.
end [ERROR]: Failed to call psconvert
end [ERROR]: process_figures returned error 78
```

It's difficult to understand why it fails from these error messages. Two possible solutions:

1. Create the output directory if it doesn't exist
2. Better error message

**Comment by PaulWessel** [comment]
Pretty sure we never checked if the directory existed before, so not a new problem.  I guess it makes sense to create that directory.  It should be easy to do when -D is given and it does not exist.  As for plot/map in the first case, we would have to discover the last slash and anything to the left is the directory.

Sounds OK to everyone?

**Comment by joa-quim** [comment]
Yes, seems harmless, aside from permission issues. If it fails due to that user still have a good ps file but harder to know where when in modern.

**Comment by PaulWessel** [comment]
Working on this.  The **-D** situation is simple and working.  For modern mode with plot/file as output then I think the session will need to make the dir.

**Comment by seisman** [comment]
I think we can declare it as a bug and fix it in 6.0.1

**Comment by PaulWessel** [comment]
Yes, all done, running tests first before PR.
